irods_testing_environment/irods_setup.py

```diff
@@ -350,8 +350,16 @@ def restart_rsyslog(container):
         if ec != 0:
             logging.info(f'[{container.name}] failed to kill rsyslogd')
 
-        ec = execute.execute_command(container, rsyslog_bin_path)
-        if ec != 0:
+        num_attempts = 0
+        while True:
+            ec = execute.execute_command(container, rsyslog_bin_path)
+            if ec != 0:
+                num_attempts += 1
+            is_alive = execute.execute_command(container, f'pgrep {os.path.basename(rsyslog_bin_path)}')
+            if ec == 0 or num_attempts >= 3 or is_alive == 0:
+                break
+
+        if num_attempts >= 3:
```

Comment:
Let's pull the `3` out into a variable (e.g. `MAX_NUMBER_OF_ATTEMPTS`?).

Consider adding some log statements to indicate what is happening.

Let's leave a comment explaining that this is temporary.

@alanking Any thoughts?

Reply:
Yes, I would definitely leave a comment indicating why this is necessary.

I would also like a comment explaining why, even though line 355 could have failed, we still check for a running rsyslog instance. It's because the OS could be (re)starting it on its own in the background, yes?

Reply:
Should be good. I had some cases where rsyslog would come up by itself(?), so those checks are for that case.

Logs added around those lines should hopefully make the intent a bit clearer.